src/Text/Pandoc/Readers/Markdown.hs

```diff
@@ -1741,6 +1742,41 @@ source = do
 linkTitle :: MarkdownParser String
 linkTitle = quotedTitle '"' <|> quotedTitle '\''
 
+wikilinkstuff :: MarkdownParser (F Inlines, String)
+wikilinkstuff = do
+  string "[["
+  (_, raw) <- withRaw $ charsInBalancedBrackets 2
+  guard $ not $ null raw
+  let deunder c = if c == '_' then ' ' else c
+      deiki s = case elemIndices '/' s of
+                  [] -> map deunder s
+                  xs -> map deunder (drop ((last xs) + 1) s)
+  case elemIndex '|' raw of
+    Nothing -> do
+      let rawstr = ((reverse . (drop 2) . reverse) raw)
+          srcstr = rawstr
+          labstr = deiki rawstr
+      src <- parseFromString (many anyChar) srcstr
+      lab <- parseFromString (trimInlinesF . mconcat <$> many inline) labstr
+      return (lab, src)
+    Just idx -> do
+      let rawstr = ((reverse . (drop 2) . reverse) raw)
+          srcstr = (drop (idx + 1) rawstr)
+          labstr = deiki (take idx rawstr)
+      src <- parseFromString (many anyChar) srcstr
+      lab <- parseFromString (trimInlinesF . mconcat <$> many inline) labstr
+      return (lab, src)
+
+wikilink :: MarkdownParser (F Inlines)
+wikilink = try $ do
+  guardEnabled Ext_ikiwiki_wikilinks
+  st <- getState
```

Comment:
Parts of this seem to be copy-pasted from `link`; it would be nicer to extract this in a separate function.

``` haskell
ensureUnnestedLinks :: MarkdownParser a -> MarkdownParser a
```

or something similar.